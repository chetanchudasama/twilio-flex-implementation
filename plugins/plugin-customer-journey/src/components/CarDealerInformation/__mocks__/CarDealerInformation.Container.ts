import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
