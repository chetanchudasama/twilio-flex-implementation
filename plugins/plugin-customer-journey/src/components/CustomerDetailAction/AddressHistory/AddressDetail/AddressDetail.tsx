import { AddressItemModel, Shared } from "@common/components";
import React, { useMemo } from "react";
import {
  Button,
  CardHeader,
  Avatar,
  Typography,
  Divider,
} from "@material-ui/core";
import HomeIcon from "@material-ui/icons/Home";
import EditIcon from "@material-ui/icons/Edit";
import { AddressItemStyles } from "./AddressDetail.Styles";

export interface AddressItemProps {
  addressItem: AddressItemModel;
  onEdit: () => void;
}

export const AddressDetail: React.FC<AddressItemProps> = ({
  addressItem,
  onEdit,
}) => {
  const addressLine1 = useMemo(() => {
    const addressLine: string[] = [];
    if (addressItem.buildingNumber) {
      addressLine.push(addressItem.buildingNumber.toString());
    }
    if (addressItem.buildingName) {
      addressLine.push(addressItem.buildingName);
    }
    return addressLine.join(", ");
  }, [addressItem]);

  const addressLine2 = useMemo(() => {
    const addressLine: string[] = [];
    if (addressItem.subBuilding) {
      addressLine.push(addressItem.subBuilding);
    }
    if (addressItem.streetName) {
      addressLine.push(addressItem.streetName);
    }
    return addressLine.join(", ");
  }, [addressItem]);

  const addressLine3 = useMemo(() => {
    const addressLine: string[] = [];
    if (addressItem.town) {
      addressLine.push(addressItem.town);
    }
    if (addressItem.postcode) {
      addressLine.push(addressItem.postcode);
    }
    return addressLine.join(", ");
  }, [addressItem]);

  return (
    <AddressItemStyles item xs={12}>
      <CardHeader
        avatar={
          <Avatar color="secondary" classes={{ colorDefault: "icon" }}>
            <HomeIcon color="secondary" />
          </Avatar>
        }
        title={
          <>
            <Typography
              component="span"
              className="address-content address-line1"
            >
              {`${addressLine1}, ${addressLine2}`}
            </Typography>
            <Typography
              component="span"
              className="address-content address-line2"
            >
              {addressLine3}
            </Typography>
          </>
        }
        subheader={
          <>
            <Typography component="span" className="subheader-line">
              {addressItem.addressStatus.addressStatusName}
            </Typography>
            <Typography component="span" className="subheader-line">
              {`${addressItem.addressStatus.yearsAtAddress} years, 
            ${addressItem.addressStatus.monthsAtAddress} months`}
            </Typography>
          </>
        }
        action={
          <Button
            variant="text"
            size="small"
            color="secondary"
            className="edit-address-btn"
            onClick={onEdit}
          >
            <EditIcon color="inherit" fontSize="small" className="edit-icon" />
            Edit
          </Button>
        }
        classes={{
          root: "card-header",
          subheader: "subheader",
        }}
      />
      <Divider className="divider" />
    </AddressItemStyles>
  );
};
