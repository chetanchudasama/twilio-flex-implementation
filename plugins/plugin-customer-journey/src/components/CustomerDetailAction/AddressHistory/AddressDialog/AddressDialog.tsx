import {
  AddressItemModel,
  CustomSelect,
  ResponsiveDialog,
  Shared,
  BaseAddressModel,
  ResidentialStatusModel,
} from "@common/components";
import {
  Checkbox,
  FormControlLabel,
  Grid,
  MenuItem,
  InputLabel,
} from "@material-ui/core";
import React, { useMemo, useState } from "react";
import { v4 } from "uuid";
import { AddressDialogProps } from "./AddressDialog.Props";
import {
  AddressDialogStyles,
  ResponsiveDialogContentStyles,
} from "./AddressDialog.Styles";
import Loading from "../../../LoadingComponent/Loading";
import { AddressSearch } from "../../../AddressSearch/AddressSearch";

export const AddressDialog: React.FC<AddressDialogProps> = ({
  activeAddress,
  open,
  onClose,
  onDelete,
  onSave,
  loading,
  isEdit,
}) => {
  const [openConfirmation, setOpenConfirmation] = useState<boolean>(false);
  const [errors, setErrors] = useState<Record<string, string[]>>({});
  const [address, setAddress] = useState<AddressItemModel>(activeAddress);

  const isAddressExist: boolean = useMemo(() => {
    return !!(
      address.postcode ||
      address.buildingName ||
      address.subBuilding ||
      address.buildingNumber ||
      address.streetName ||
      address.town
    );
  }, [address]);

  const onChangeAddressDetail = (event: any, propertyName: string) => {
    const {
      target: { value, checked },
    } = event;

    if (propertyName === "isPrimaryAddress") {
      setAddress({
        ...address,
        [propertyName]: checked,
      });
      return;
    }
    if (propertyName === "residentialStatus") {
      const residentialStatus: ResidentialStatusModel | undefined =
        Shared.residentialStatuses.find(
          (s: ResidentialStatusModel) => s.id === Number(value)
        );
      setAddress({
        ...address,
        addressStatus: {
          ...address.addressStatus,
          addressStatusId: residentialStatus ? residentialStatus.id : -1,
          addressStatusName: residentialStatus ? residentialStatus.name : "",
        },
      });
    } else {
      setAddress({
        ...address,
        addressStatus: {
          ...address.addressStatus,
          [propertyName]: value,
        },
      });
    }
  };

  const onSaveAddressDetail = () => {
    const errorObj: Record<string, string[]> = {};
    if (address.addressStatus.addressStatusId <= 0) {
      errorObj.residentialStatus = ["The residential status is required."];
    }

    if (
      address.addressStatus.yearsAtAddress === 0 &&
      address.addressStatus.monthsAtAddress === 0
    ) {
      errorObj.residentialPeriod = ["Residential period is required."];
    }

    if (!isAddressExist) {
      errorObj.customerAddress = ["Address of customer is required."];
    }

    setErrors(errorObj);

    if (Object.keys(errorObj).length === 0) {
      onSave(address);
    }
  };

  const handleUpdateAddress = (baseAddressModel: BaseAddressModel) => {
    setAddress({
      ...address,
      buildingName: baseAddressModel.buildingName,
      buildingNumber: baseAddressModel.buildingNumber,
      postcode: baseAddressModel.postcode,
      streetName: baseAddressModel.streetName,
      subBuilding: baseAddressModel.subBuilding,
      town: baseAddressModel.town,
    });
  };

  return (
    <AddressDialogStyles item xs={12}>
      {open && (
        <ResponsiveDialog
          title={isEdit ? "Edit Address" : "Add Address"}
          open={open}
          onCancel={onClose}
          onConfirm={onSaveAddressDetail}
          maxWidth="sm"
          showActionButtons
          showDeleteButton={isEdit}
          onDelete={() => {
            setOpenConfirmation(true);
          }}
          okText="Save Address"
          cancelText="Cancel"
          deleteText="Delete Address"
        >
          <ResponsiveDialogContentStyles item xs={12}>
            <Grid
              container
              direction="row"
              justify="space-between"
              alignItems="center"
              spacing={8}
            >
              <Grid item xs={12} className="address-container">
                <InputLabel shrink required error={!!errors.customerAddress}>
                  Address
                </InputLabel>
                <AddressSearch
                  updateAddress={handleUpdateAddress}
                  address={isAddressExist ? activeAddress : undefined}
                />
                {!!errors.customerAddress && (
                  <p className="address-error">{errors.customerAddress}</p>
                )}
              </Grid>
              <Grid item xs={12} className="residential-status">
                <CustomSelect
                  label="Residential status"
                  onChange={(event: any) =>
                    onChangeAddressDetail(event, "residentialStatus")
                  }
                  value={address.addressStatus.addressStatusId || -1}
                  isRequired
                  errors={errors.residentialStatus}
                >
                  <MenuItem value={-1}>
                    <em>Select a status</em>
                  </MenuItem>
                  {Shared.residentialStatuses.map(
                    (status: ResidentialStatusModel) => (
                      <MenuItem key={v4()} value={status.id}>
                        {status.name}
                      </MenuItem>
                    )
                  )}
                </CustomSelect>
              </Grid>
              <Grid item xs={6} className="residential-period">
                <CustomSelect
                  label="Years"
                  onChange={(event: any) =>
                    onChangeAddressDetail(event, "yearsAtAddress")
                  }
                  value={address.addressStatus.yearsAtAddress}
                  isRequired
                  errors={errors.residentialPeriod}
                >
                  {Array.from(Array(25).keys()).map((n: number) => (
                    <MenuItem key={v4()} value={n}>
                      {`${n} Years`}
                    </MenuItem>
                  ))}
                </CustomSelect>
              </Grid>
              <Grid item xs={6} className="residential-period">
                <CustomSelect
                  label="Months"
                  onChange={(event: any) =>
                    onChangeAddressDetail(event, "monthsAtAddress")
                  }
                  value={address.addressStatus.monthsAtAddress}
                  isRequired
                  errors={errors.residentialPeriod}
                >
                  {Array.from(Array(12).keys()).map((n: number) => (
                    <MenuItem key={v4()} value={n}>
                      {`${n} Months`}
                    </MenuItem>
                  ))}
                </CustomSelect>
              </Grid>
              <Grid item xs={12}>
                <FormControlLabel
                  control={
                    <Checkbox
                      checked={address.isPrimaryAddress}
                      onChange={(event) =>
                        onChangeAddressDetail(event, "isPrimaryAddress")
                      }
                      value="isPrimaryAddress"
                    />
                  }
                  label="This is their current address"
                />
              </Grid>
            </Grid>
            {loading && <Loading />}
          </ResponsiveDialogContentStyles>
        </ResponsiveDialog>
      )}
      {openConfirmation && (
        <ResponsiveDialog
          open={openConfirmation}
          maxWidth="xs"
          onConfirm={onDelete}
          onCancel={() => setOpenConfirmation(false)}
          showActionButtons
          okText="Yes"
          cancelText="No"
          isConfirmationDialog
        >
          <div className="confirmation-message">
            Are you sure you want to delete this address?
          </div>
        </ResponsiveDialog>
      )}
    </AddressDialogStyles>
  );
};
