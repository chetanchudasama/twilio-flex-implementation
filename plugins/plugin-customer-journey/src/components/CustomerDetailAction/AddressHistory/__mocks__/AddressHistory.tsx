import React from "react";
import { AddressHistoryProps } from "../AddressHistory";

const AddressHistory: React.FC<AddressHistoryProps> = (props) => (
  <div className="address-history" />
);

export default AddressHistory;
