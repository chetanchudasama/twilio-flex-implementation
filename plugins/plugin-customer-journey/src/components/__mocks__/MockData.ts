import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
