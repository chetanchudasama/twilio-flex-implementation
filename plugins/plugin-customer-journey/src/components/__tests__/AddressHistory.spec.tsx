import React from "react";
import { Manager } from "@twilio/flex-ui";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { create } from "react-test-renderer";
import { AddressItemModel } from "@common/components";
import { AddressDetail } from "../CustomerDetailAction/AddressHistory/AddressDetail/AddressDetail";
import {
  StateToProps,
  DispatchToProps,
} from "../CustomerDetailAction/AddressHistory/AddressHistory.Props";
import { AddressHistory } from "../CustomerDetailAction/AddressHistory/AddressHistory";
import { flexManagerMock } from "../__mocks__/MockData";
import { AddressDialog } from "../CustomerDetailAction/AddressHistory/AddressDialog/AddressDialog";
import { showErrorMessage, showMessage } from "../../Notifications";

Manager.getInstance = flexManagerMock;

const mockUpdateBaseApplicationDetail = jest.fn();

jest.mock("../../services/application.service", () => ({
  useApplicationService: jest.fn(() => ({
    updateBaseApplicationDetail: mockUpdateBaseApplicationDetail,
  })),
}));

jest.mock("../../Notifications", () => {
  return {
    showErrorMessage: jest.fn(() => {}),
    showMessage: jest.fn(() => {}),
    CustomNotificationType: { SuccessNotification: "successNotification" },
  };
});

type Props = StateToProps & DispatchToProps;

const address: AddressItemModel[] = [
  {
    postcode: "NP6 8RF",
    buildingName: "Parsley",
    subBuilding: "Parsley-9",
    buildingNumber: "114",
    streetName: "48 Bullwood Rd",
    town: "ST ARVANS",
    addressStatus: {
      addressStatusId: 0,
      addressStatusName: "Home Owner",
      yearsAtAddress: 2,
      monthsAtAddress: 27,
    },
    isPrimaryAddress: true,
  },
  {
    postcode: "NN6 2XS",
    buildingName: "Grange",
    subBuilding: "Grange-1",
    buildingNumber: "110",
    streetName: "43 Buckingham Rd",
    town: "THORNBY",
    addressStatus: {
      addressStatusId: 2,
      addressStatusName: "Tenant",
      yearsAtAddress: 1,
      monthsAtAddress: 12,
    },
    isPrimaryAddress: false,
  },
  {
    postcode: "YO4 3XA",
    buildingName: "Lilac",
    subBuilding: "Lilac-3",
    buildingNumber: "112",
    streetName: "114 Long Street",
    town: "MILLINGTON",
    addressStatus: {
      addressStatusId: 3,
      addressStatusName: "Private Tenant",
      yearsAtAddress: 0,
      monthsAtAddress: 11,
    },
    isPrimaryAddress: false,
  },
];

const props: Props = {
  applicationId: 1,
  currentAddress: address[0],
  previousAddressList: [address[1], address[2]],
  setCustomerDetails: jest.fn(),
};

describe("AddressHistory", () => {
  it("renders correctly", () => {
    const wrapper = create(<AddressHistory {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders current address and previous addresses correctly in UI", () => {
    const wrapper = create(<AddressHistory {...props} />).root;

    expect(
      wrapper.findAllByProps({
        className: "address-header",
      })[1].props.children
    ).toContain("Current Address");
    expect(
      wrapper.findAllByProps({
        className: "address-header",
      })[2].props.children
    ).toContain("Previous Addresses");
  });

  it("renders address history item correctly in UI", () => {
    const wrapper = mount(<AddressHistory {...props} />);

    expect(wrapper.find(AddressDetail)).toHaveLength(
      [props.currentAddress, ...props.previousAddressList].length
    );
  });

  it("should open address dialog, when add address button is clicked", async () => {
    const wrapper = mount(<AddressHistory {...props} />);

    wrapper.find(".add-address-btn").at(0).simulate("click");
    wrapper.update();

    expect(wrapper.find(AddressDialog).exists()).toBeTruthy();
  });

  it("should open address dialog, when edit address button is clicked", async () => {
    const wrapper = mount(<AddressHistory {...props} />);

    act(() => {
      wrapper.find(AddressDetail).at(0).props().onEdit();
    });
    wrapper.update();

    expect(wrapper.find(AddressDialog).exists()).toBeTruthy();
  });

  it("should close address dialog, when close button is clicked from AddressDialog", async () => {
    const wrapper = mount(<AddressHistory {...props} />);

    wrapper.find(".add-address-btn").at(0).simulate("click");
    wrapper.update();

    expect(wrapper.find(AddressDialog).exists()).toBeTruthy();

    act(() => {
      wrapper.find(AddressDialog).props().onClose();
    });
    wrapper.update();

    expect(wrapper.find(AddressDialog).exists()).toBeFalsy();
  });

  it("should delete address, when delete button is clicked from AddressDialog", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.resolve(undefined)
    );

    const wrapper = mount(<AddressHistory {...props} />);

    act(() => {
      wrapper.find(AddressDetail).at(0).props().onEdit();
    });
    wrapper.update();
    expect(wrapper.find(AddressDialog).exists()).toBeTruthy();

    wrapper.find(AddressDialog).props().onDelete();
    wrapper.update();

    await act(async () => {
      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalled();
      wrapper.update();
    });

    expect(showMessage).toHaveBeenCalledWith(
      "successNotification",
      "Address detail deleted successfully!"
    );
  });

  it("should display error message, if delete address detail API returns an error", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.reject()
    );

    const wrapper = mount(<AddressHistory {...props} />);

    act(() => {
      wrapper.find(AddressDetail).at(0).props().onEdit();
    });
    wrapper.update();

    expect(wrapper.find(AddressDialog).exists()).toBeTruthy();

    wrapper.find(AddressDialog).props().onDelete();
    wrapper.update();

    await act(async () => {
      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalled();
      wrapper.update();
    });

    expect(showErrorMessage).toHaveBeenCalledWith(
      "Something went wrong, please try again!",
      "",
      true
    );
  });

  it("should update address, when edit address button is clicked from AddressDialog", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.resolve(undefined)
    );

    const wrapper = mount(<AddressHistory {...props} />);

    act(() => {
      wrapper.find(AddressDetail).at(0).props().onEdit();
    });
    wrapper.update();

    expect(wrapper.find(AddressDialog).exists()).toBeTruthy();

    await act(async () => {
      wrapper.find(AddressDialog).props().onSave(address[1]);
      wrapper.update();

      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalled();
    });

    expect(showMessage).toHaveBeenCalledWith(
      "successNotification",
      "Address detail updated successfully!"
    );
  });

  it("should show error message, if update address detail API returns an error", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.reject()
    );

    const wrapper = mount(<AddressHistory {...props} />);

    act(() => {
      wrapper.find(AddressDetail).at(0).props().onEdit();
    });
    wrapper.update();

    expect(wrapper.find(AddressDialog).exists()).toBeTruthy();

    await act(async () => {
      wrapper.find(AddressDialog).props().onSave(address[1]);
      wrapper.update();

      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalled();
    });

    expect(showErrorMessage).toHaveBeenCalledWith(
      "Something went wrong, please try again!",
      "",
      true
    );
  });

  it("should show No Address found message, if addresses are not available", async () => {
    props.currentAddress = null;
    props.previousAddressList = [];
    const wrapper = mount(<AddressHistory {...props} />);

    expect(wrapper.find(AddressDetail).exists()).toBeFalsy();

    expect(wrapper.html()).toContain("No Address found.");
  });
});
