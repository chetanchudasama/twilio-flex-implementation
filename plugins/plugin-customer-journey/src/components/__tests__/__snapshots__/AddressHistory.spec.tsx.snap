// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressHistory renders correctly 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-1o0mgdm"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 address-history-header"
    >
      Address History
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 paper-container"
    >
      <div
        className="MuiPaper-root-98 MuiPaper-elevation1-101 MuiPaper-rounded-99 paper"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
          >
            <p
              className="MuiTypography-root-125 MuiTypography-body1-134 address-header"
            >
              Current Address
            </p>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-5st4sl"
          >
            <div
              className="MuiCardHeader-root-161 card-header"
            >
              <div
                className="MuiCardHeader-avatar-162"
              >
                <div
                  className="MuiAvatar-root-167 MuiAvatar-colorDefault-168 icon"
                  color="secondary"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-170 MuiSvgIcon-colorSecondary-172"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="MuiCardHeader-content-164"
              >
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiCardHeader-title-165"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 address-content address-line1"
                  >
                    114, Parsley, Parsley-9, 48 Bullwood Rd
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 address-content address-line2"
                  >
                    ST ARVANS, NP6 8RF
                  </span>
                </span>
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiTypography-colorTextSecondary-158 MuiCardHeader-subheader-166 subheader"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    Home Owner
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    2 years, 
            27 months
                  </span>
                </span>
              </div>
              <div
                className="MuiCardHeader-action-163"
              >
                <button
                  className="MuiButtonBase-root-205 MuiButton-root-179 MuiButton-text-181 MuiButton-textSecondary-183 MuiButton-flat-184 MuiButton-flatSecondary-186 MuiButton-sizeSmall-202 edit-address-btn"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-180"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-170 MuiSvgIcon-fontSizeSmall-177 edit-icon"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    Edit
                  </span>
                  <span
                    className="MuiTouchRipple-root-213"
                  />
                </button>
              </div>
            </div>
            <hr
              className="MuiDivider-root-208 divider"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root-98 MuiPaper-elevation1-101 MuiPaper-rounded-99 paper"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
          >
            <p
              className="MuiTypography-root-125 MuiTypography-body1-134 address-header"
            >
              Previous Addresses
            </p>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-5st4sl"
          >
            <div
              className="MuiCardHeader-root-161 card-header"
            >
              <div
                className="MuiCardHeader-avatar-162"
              >
                <div
                  className="MuiAvatar-root-167 MuiAvatar-colorDefault-168 icon"
                  color="secondary"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-170 MuiSvgIcon-colorSecondary-172"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="MuiCardHeader-content-164"
              >
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiCardHeader-title-165"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 address-content address-line1"
                  >
                    110, Grange, Grange-1, 43 Buckingham Rd
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 address-content address-line2"
                  >
                    THORNBY, NN6 2XS
                  </span>
                </span>
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiTypography-colorTextSecondary-158 MuiCardHeader-subheader-166 subheader"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    Tenant
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    1 years, 
            12 months
                  </span>
                </span>
              </div>
              <div
                className="MuiCardHeader-action-163"
              >
                <button
                  className="MuiButtonBase-root-205 MuiButton-root-179 MuiButton-text-181 MuiButton-textSecondary-183 MuiButton-flat-184 MuiButton-flatSecondary-186 MuiButton-sizeSmall-202 edit-address-btn"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-180"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-170 MuiSvgIcon-fontSizeSmall-177 edit-icon"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    Edit
                  </span>
                  <span
                    className="MuiTouchRipple-root-213"
                  />
                </button>
              </div>
            </div>
            <hr
              className="MuiDivider-root-208 divider"
            />
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-5st4sl"
          >
            <div
              className="MuiCardHeader-root-161 card-header"
            >
              <div
                className="MuiCardHeader-avatar-162"
              >
                <div
                  className="MuiAvatar-root-167 MuiAvatar-colorDefault-168 icon"
                  color="secondary"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-170 MuiSvgIcon-colorSecondary-172"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="MuiCardHeader-content-164"
              >
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiCardHeader-title-165"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 address-content address-line1"
                  >
                    112, Lilac, Lilac-3, 114 Long Street
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 address-content address-line2"
                  >
                    MILLINGTON, YO4 3XA
                  </span>
                </span>
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiTypography-colorTextSecondary-158 MuiCardHeader-subheader-166 subheader"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    Private Tenant
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    0 years, 
            11 months
                  </span>
                </span>
              </div>
              <div
                className="MuiCardHeader-action-163"
              >
                <button
                  className="MuiButtonBase-root-205 MuiButton-root-179 MuiButton-text-181 MuiButton-textSecondary-183 MuiButton-flat-184 MuiButton-flatSecondary-186 MuiButton-sizeSmall-202 edit-address-btn"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-180"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-170 MuiSvgIcon-fontSizeSmall-177 edit-icon"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    Edit
                  </span>
                  <span
                    className="MuiTouchRipple-root-213"
                  />
                </button>
              </div>
            </div>
            <hr
              className="MuiDivider-root-208 divider"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 add-btn-container"
    >
      <button
        className="MuiButtonBase-root-205 MuiButton-root-179 MuiButton-contained-190 MuiButton-containedSecondary-192 MuiButton-raised-193 MuiButton-raisedSecondary-195 add-address-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-180"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-170 MuiSvgIcon-fontSizeSmall-177 add-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Add Address
        </span>
        <span
          className="MuiTouchRipple-root-213"
        />
      </button>
    </div>
  </div>
</div>
`;
